---
title: "regulaTER: an R package for analysis of transposable elements in accessible regions"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{introduction}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```
.
This repo is currently under review. Citation information will be provided as soon as our work is accepted.

### What is this package used for? 
This package is used for analysis of repeat elements in the genome, more specifically transposable elements, and their association with accessible chromatin and gene expression regulatory elements. As input, it requires DNA accessibility data, such as that produced by ATAC-seq, and analyzed by an appropriate pipeline, a RepeatMasker file including information for genomic repeats, BED files describing the gene-related contexts of all genomic regions, and a list of differentially expressed genes in condition of choice compared to controls and their genomic coordinates.


#### What are the dependencies for regulaTER ?
1. [R](https://www.r-project.org/) version should be version 3.5+
2. While using R programming, we recommend the use of [Rstudio](https://www.rstudio.com/products/rstudio/download/) for convenient use and understanding of the functions of regulaTER.
3. [Bedtools](https://bedtools.readthedocs.io/en/latest/content/installation.html) is required to be installed and in your PATH environmental variable.
4. [HOMER](http://homer.ucsd.edu/homer/introduction/install.html) is required to be installed and in your PATH environmental variable.
5. [devtools](https://cran.r-project.org/web/packages/devtools/readme/README.html) is required to install regulaTER.
6. regulaTER utilizes functions from the following R packages, which have to be installed in your R library. You may visit the following websites to install them easily: 
    - [dplyr](https://dplyr.tidyverse.org/)
    - [GenomicRanges](https://bioconductor.org/packages/release/bioc/html/GenomicRanges.html)
    - [biomartr](https://cran.r-project.org/web/packages/biomartr/readme/README.html)
    - [marge](https://robertamezquita.github.io/marge/index.html)


### How to install this R package ?

```{r, eval=FALSE}

library(devtools)
devtools::install_github("karakulahg/regulaTER")

```

### How does it work?

For the analysis of narrowPeak peak files generated by MACS2, follow the steps outlined below.

1. Install and load the following libraries:
```{r}
suppressPackageStartupMessages({

library(regulaTER)
library(biomartr)
library(GenomicRanges)
library(dplyr)
library(biomaRt)
library(marge)
library(ChIPseeker)
  
})

```

2. Read the repeat track information for your genome of interest in RepeatMasker file format, as obtained from the [RepeatMasker database](https://www.repeatmasker.org/genomicDatasets/RMGenomicDatasets.html), as well as your genomic sequencing peak information, generated by MACS2 and annotated by ChIPseeker, in one of narrowPeak, broadPeak, or summits formats. For details on the format of annotated BED files, see "ChIPseeker Annotations Compatible with ShufflePeaks and TEAR Functions" below these instructions.

```{r}
options(warn = -1)
raw.rmsk<-biomartr::read_rm("/home/nazmiye/Desktop/ibg/test_ToolX/hg38/hg38.fa.out.gz")
head(raw.rmsk)

peak <-readPeakFile("/home/nazmiye/Desktop/ibg/ZGA-SRP112718-ToolX/ATACSeq-Pipelines/AnnotatedNarrowPeaks/Annotated_FinalPeakListSRX3884851_SRX3884852.narrowPeak.filt.gz.tsv")

# explicitly identify the column stating the distance of the summit nucleotide from the peak start nucleotide (for narrowPeak files only)
names(mcols(peak))[8] <- "summit"
head(peak)
```

3. Using the Table Browser tool of UCSC Genome Browser, or another similar method, generate BED files for the following genomic regions, as compatible with ChIPseeker annotations. Promoter region is defined as 3000 bases upstream of the gene region, while Downstream region is defined as 3000 bases downstream. The genomeSizePath should point to a file with two columns, with the first column corresponding to chromosome names in order, and the second column corresponding to chromosome size. Details on how to use Table Browser to obtain the BED files is included below the instructions for regulaTER. For the UCSC human and mouse genome assemblies hg38, hg19, and mm10, the pre-generated region files are available for download via the GitHub repository [regulaTER-regions](https://github.com/karakulahg/regulaTER-regions). 

```{r}
base_path <- "/home/nazmiye/Desktop/hg38_regions"

```


```{r}
pathList <- list("Promoter" = paste0(base_path, "/hg38_promoter_complement.bed"),
                 "Exon" =  paste0(base_path, "/hg38_exons_complement.bed"),
                 "Intron" =  paste0(base_path, "/hg38_introns_complement.bed"),
                 "5UTR" =  paste0(base_path, "/hg38_5prime_complement.bed"),
                 "3UTR" =  paste0(base_path, "/hg38_3prime_complement.bed"),
                 "Downstream" =  paste0(base_path, "/hg38_downstream_complement.bed"),
                 "genomeSizePath" =  paste0(base_path, "/hg38.chrom.sizes"))


```
4. With the previously generated peak file and repeat masker objects, as well as the list of file paths to BED files defining genomic regions and the file with chromosome sizes, calculate the fold enrichment and p-values of repeat elements associated with input peak regions (Transposable Elements in Accessible Regions - TEAR). If a broadPeak file is used, the format should be changed as "broad", and minoverlap should be given an integer value, defining how many minimum bases should a peak and a repeat element overlap before they are considered associated. Higher shuffle numbers give better results, but increase operation time.

```{r}

enrich.peak <- TEAR(inputPeakFile = peak, pathList = pathList, numberOfShuffle = 2, repeatMaskerFile = raw.rmsk, format="narrow", minoverlap=0L, alternative = "greater", minobserved = 10)

head(enrich.peak$RepeatName[order(enrich.peak$RepeatName$p.adjust.value), ])

```

5. Using the results of the TEAR function, the peak file and repeat masker objects used as the input for TEAR, and a user provided data frame of differentially expressed genes and their genomic intervals (such as those obtained from ensembl's BioMart or UCSC's Genome Browser), the structure of which is described under the function documentation, calculate the fold enrichment and p-values of accessible region enriched repeat elements located in input promoter regions (DEG Associated Transposable Elements - DATE). Higher shuffle numbers give better results, but increase operation time. Distance value defines desired length of promoter region from transcription start site. Longer distances cover more distal regulatory elements, but reduce precision of results.

```{r}

# use getInterval to generate interval ranges from gene list
input <- "~/Desktop/ibg/test_ToolX/ToolX-inputs/DE_genenames.txt"
dataset <- "hsapiens_gene_ensembl"

genes <- regulaTER::getInterval(input, dataset)
head(genes)


IdDEGRepeats <- DATE(enrichTEARResult = enrich.peak, peaks = peak, rmsk = raw.rmsk, genes = genes, alternative = "greater" ,numberOfShuffle = 2, minobserved = 10, distance = 100000)

head(IdDEGRepeats[order(IdDEGRepeats$p.adjust.value), ])
```

6. Using the output of the DATE function, as well as the peak, repeat masker, and genes object used in the same function, identify which genomic motifs are enriched in promoter associated TEAR regions. This function requires a local installation of HOMER, available on the [HOMER website](http://homer.ucsd.edu/homer/). In case the marge API has difficulty locating HOMER tools, refer to the [marge repository](https://github.com/robertamezquita/marge).

```{r, eval = FALSE}

check_homer()

get_homer_bin()

# replace the homerPath argument with the output of get_homer_bin, removing the "/bin/" suffix at the end

FindMotifs(df = IdDEGRepeats, repeatMaskerFile = raw.rmsk, peak = peak, distance = 100000, genes = genes, genome = "hg38", outDir = "../test/", homerPath = "~/path/to/homer/./", type = "linkedRepeats", topRepeats = T)
```

Alternately, the repeatName object in the output of TEAR can be provided instead of the output of DATE. In this case, the distance and genes parameters can be omitted.

```{r, eval = FALSE}
FindMotifs(df = enrich.peak$RepeatName, repeatMaskerFile = raw.rmsk, peak = peak, genome = "hg38", outDir = "../test/", homerPath = "~/path/to/homer/./", type = "enrichPeak", topRepeats = T)
```

7. Using the output of the TEAR function, generate exploratory plots of accessible repeat regions and place them in a directory of choice. For larger datasets, increase the width and height 

```{r, eval = FALSE}
MakePlots(enrich.peak "~/output/directory", width = 4, height = 4)
```


### Analysis of broadPeak and summits files

For broadPeak files, many of the above steps can be used as is. As broad peak analysis does not identify summit nucleotide, broadPeak files lack the summit column. Therefore, the summit column is not used for the analysis, and does not have to be identified. In addition, the minimum amount of overlapping bases between the peaks and the repeat regions has to be provided as a long integer. It is recommended that the value is no larger than half the length of the shortest repeat of interest, or half the read length of the sequencing library used to identify peaks, whichever is smaller.

```{r}
options(warn = -1)
raw.rmsk<-biomartr::read_rm("/home/nazmiye/Desktop/ibg/test_ToolX/mm10/mm10.fa.out")
head(raw.rmsk)

peak <-readPeakFile("/home/nazmiye/Desktop/ibg/test_ToolX/ToolX-inputs/regulaTER-broadPeak/Annotated_Vehicle_K27Ac_IP-S1_R1_peaks.broadPeak.tsv")

head(peak)

base_path <- "/home/nazmiye/mm10_regions"

pathList <- list("Promoter" = paste0(base_path, "/mm10_promoter_complement.bed"),
                 "Exon" =  paste0(base_path, "/mm10_exons_complement.bed"),
                 "Intron" =  paste0(base_path, "/mm10_introns_complement.bed"),
                 "5UTR" =  paste0(base_path, "/mm10_5prime_complement.bed"),
                 "3UTR" =  paste0(base_path, "/mm10_3prime_complement.bed"),
                 "Downstream" =  paste0(base_path, "/mm10_downstream_complement.bed"),
                 "genomeSizePath" =  paste0(base_path, "/mm10_chrom.sizes"))

enrich.peak <- TEAR(inputPeakFile = peak, pathList = pathList, numberOfShuffle = 2, repeatMaskerFile = raw.rmsk, format="broad", minoverlap=10L, alternative = "greater", minobserved = 10)

head(enrich.peak$RepeatName[order(enrich.peak$RepeatName$p.adjust.value), ])

``` 

Files in the summits format only define the summit nucleotide for each identified peak, and use the same analysis steps as for narrowPeak files, with the exception that the summit column does not have to be identified.

### Using the UCSC Table Browser to Obtain Genomic Regions Compatible with ChIPseeker Regions

ChIPseeker regions are divided into seven categories. In order of priority, these are promoter, coding exon, intron, 5' UTR, 3' UTR, downstream regions, and intergenic regions. Overlaps are categorized based on the order of priority when annotating genomic regions. As the shuffling of peak regions are performed in the same category of region as they are originally found in, the intervals for these regions must be supplied by the user. If you are working with a genome with known gene annotations available on the UCSC Genome Browser, you can obtain these files using the Table Browser tool.

Under genome and assembly, select the organism and genome assembly of choice, also to be used with ChIPseeker to annotate the peak regions. For region, select "genome", and for output format, choose "BED - browser extensible data". Fill the output file field to download the resulting file to your system.

On the next screen, you can further select the regions your file will have. For promoter and downstream regions, select "Upstream by" or "Downstream by", respectively, in both cases using 3000 bases as the region length. Intergenic regions need not be downloaded, as any region not falling under another category are included in intergenic shuffles.

In the current version of regulaTER, the complement intervals of the regions must be provided for each object in the list. They use the same format as the obtained BED files, but cover all genomic regions not found in the category. The user can generate these files using another genome arithmetic tool, such as bedtools complement (-i BED -g genome), available on the [bedtools Suite](https://bedtools.readthedocs.io/en/latest/index.html).

The user will also require the full lengths of the chromosomes included in the annotation. These are available for each genome found in UCSC Genome Browser under Downloads / Genome Data, with the name "[assembly].chrom.sizes", under the link named "Genome sequence files and select annotations (2bit, GTF, GC-content, etc)".

If the user is using annotations not available on the UCSC Genome Browser, they can manually generate these files using the tool of their choice, as long as gene and exon annotations are already available in the BED format.

### ChIPseeker Annotations Compatible with ShufflePeaks and TEAR Functions

As of MACS2 (version 2.1.2) and ChIPseeker (version 1.22.1), a narrowPeak format BED file generated by MACS2 and annotated by ChIPseeker has 19 columns. The first eleven columns of the file correspond to the original BED file, with the remaining columns added by ChIPseeker. Only the first give columns, the summit column and the annotation fields are used during regulaTER's shufflePeak function. seqnames identifies the chromosome the peak is located in, start and end columns identify the 5' and 3' of the interval on the sense strand, width identifies the length of the interval, strand identifies the strand information of the peak, if applicable. The "summit" column (by default the 13th column of the BED file) identifies the distance of the summit nucleotide from the interval start site. The "annotation" column identifies the peak's relationship to nearby genes, and can be one of Promoter, Exon, Intron, 5' UTR, 3' UTR, Downstream, and Distal Intergenic, with further annotation specifying distance from TSS or TES, as well as which exon or intron, if applicable. If the user desires to supply their own annotated interval regions, the minimum required columns for the GRanges object used in regulaTER functions are named "seqnames", "ranges" ("start" and "end"), "strand", "summit" and "annotation".

BED files in broadPeak or summits have one fewer column than those in the narrowPeak format, as they lack the summit column.


### Session Information

```{r}

sessionInfo()

```

  
